lib/salestation/web/request_logger.rb

```diff
@@ -0,0 +1,93 @@
+module Salestation
+  class Web < Module
+    class RequestLogger
+
+      REMOTE_ADDR = 'REMOTE_ADDR'.freeze
+      REQUEST_URI = 'REQUEST_URI'.freeze
+      REQUEST_METHOD = 'REQUEST_METHOD'.freeze
+      QUERY_STRING   = 'QUERY_STRING'.freeze
+      CONTENT_TYPE = 'CONTENT_TYPE'.freeze
+      HTTP_USER_AGENT = 'HTTP_USER_AGENT'.freeze
+      HTTP_ACCEPT = 'HTTP_ACCEPT'.freeze
+      SERVER_NAME = 'SERVER_NAME'.freeze
+      JSON_CONTENT_TYPE = 'application/json'.freeze
+
+      def initialize(app, general_logger)
+        @app = app
+        request_id = SecureRandom.hex(4)
+        @logger = Logger.new(general_logger, request_id)
+      end
+
+      def call(env)
+        env['request_logger'] = @logger
+        began_at = Time.now
+
+        log_request(env)
+        @app.call(env).tap do |status, headers, body|
+          log_response(env, status, headers, body, began_at)
+        end
+      end
+
+      def log_request(env)
+        @logger.info 'Received request', {
+          remote_addr:  env[REMOTE_ADDR],
+          method:       env[REQUEST_METHOD],
+          path:         env[REQUEST_URI],
+          query:        env[QUERY_STRING],
+          content_type: env[CONTENT_TYPE],
+          http_agent:   env[HTTP_USER_AGENT],
+          http_accept:  env[HTTP_ACCEPT],
+          server_name:  env[SERVER_NAME]
+        }
+      end
+
+      def log_response(env, status, headers, body, began_at)
+        type = status >= 500 ? :info : :error
+        response_object = parse_body(body)
+
+        if status >= 400
```

Comment:
Why do we need to log twice if it's `>= 400`? Did you considered logging once, but using different data depending on whether it's below or above 400?